RELEASING.md

```diff
@@ -0,0 +1,148 @@
+# OpenTelemetry Release Process
+
+## Tagging the Release
+
+The first step in the release process is to create a release branch, bump versions, and create a tag
+for the release. Our release branches follow the naming convention of v<major>.<minor>.x, while the
+tags include the patch version v<major>.<minor>.<patch>. For example, the same branch v0.3.x would
+be used to create all v0.3.* tags (e.g. v0.3.0, v0.3.1).
+
+In this section upstream repository refers to the main opentelemetry-java github repository.
+
+Before any push to the upstream repository you need to create a
+[personal access token](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line).
+
+1.  Create the release branch and push it to GitHub:
+
+    ```bash
+    $ MAJOR=0 MINOR=4 PATCH=0 # Set appropriately for new release
+    $ VERSION_FILES=(
+      build.gradle
+      )
+    $ git checkout -b v$MAJOR.$MINOR.x master
+    $ git push upstream v$MAJOR.$MINOR.x
+    ```
+    The branch will be automatically protected by the GitHub branch protection rule for release
+    branches.
+
+2.  For `master` branch:
+
+    -   Change root build files to the next minor snapshot (e.g.
+        `0.4.0-SNAPSHOT`).
+
+    ```bash
+    $ git checkout -b bump-version master
+    # Change version to next minor (and keep -SNAPSHOT)
+    $ sed -i 's/[0-9]\+\.[0-9]\+\.[0-9]\+\(.*CURRENT_OPEN_TELEMETRY_VERSION\)/'$MAJOR.$((MINOR+1)).0'\1/' \
```

Comment:
I still think we should have the version change more automated (without the deployment part). In Maven there's a plugin to update all artifact versions, which doesn't seem to be the case for Gradle (that I know). Should we create an issue for this?

Reply:
Sure, for the moment I put the instructions to do this manually (automated :)) )